distribution/debian/packages.bzl

```diff
@@ -0,0 +1,109 @@
+load("@rules_pkg//:pkg.bzl", "pkg_deb", "pkg_tar")
+
+GLIBC_MIN_VERSION = "2.27"
+
+def envoy_pkg_deb(
+        name,
+        data,
+        homepage = "https://www.envoyproxy.io/",
+        description = "Envoy built for Debian/Ubuntu",
+        preinst = "//distribution/debian:preinst",
+        supported_distributions = "buster bullseye bionic focal hirstute impish",
+        architecture = select({
+            "//bazel:x86": "amd64",
+            "//conditions:default": "arm64",
+        }),
+        depends = [
+            "libc6 (>= %s)" % GLIBC_MIN_VERSION,
+        ],
+        version = None,
+        maintainer = None,
+        **kwargs):
+    """Wrapper for `pkg_deb` with Envoy defaults"""
+    pkg_deb(
+        name = "%s.deb" % name,
+        architecture = architecture,
+        data = data,
+        depends = depends,
+        description = description,
+        distribution = supported_distributions,
+        homepage = homepage,
+        maintainer = maintainer,
+        package = name,
+        version = version,
+        preinst = preinst,
+        **kwargs
+    )
+
+def envoy_pkg_debs(name, version, release_version, maintainer, bin_files = ":envoy-bin-files", config = ":envoy-config"):
+    """Package the Envoy .debs with their .changes files.
+
+    Packages are created for the version *and* the release version, eg
+
+    - envoy_1.21.0_amd64.deb
+    - envoy-1.21_1.21.0_amd64.deb
+
+    This way packages are available for both "envoy" and "envoy-1.21" in package managers.
+    """
+
+    # generate deb data for all packages
+    pkg_tar(
+        name = "deb-data",
+        srcs = [
+            "//distribution/debian:copyright",
+            config,
+            bin_files,
+        ],
+        remap_paths = {"/copyright": "/usr/share/doc/envoy/copyright"},
+    )
+
+    # generate package for this patch version
+    envoy_pkg_deb(
+        name = "envoy",
+        data = ":deb-data",
+        version = version,
+        maintainer = maintainer,
+    )
+
+    # generate package for this minor version
+    envoy_pkg_deb(
+        name = "envoy-%s" % release_version,
+        data = ":deb-data",
+        version = version,
+        conflicts = ["envoy"],
+        provides = ["envoy"],
+        maintainer = maintainer,
+    )
+
+    # TODO(phlax): Remove this hack
+    #   Due to upstream issues with `OutputGroupInfo` files from `pkg_deb`
+    #   we have to follow the real filepath of the .deb file to find the
+    #   .changes file (~artefact)
```

Comment:
where is .changes files generated/included and do we really need them? I'd rather remove this hack and go without .changes.

Reply:
changes files are required to sign debs - i think its really not something we can do without

its generated by the `pkg_deb` rule and previously worked ~as expected - im hoping it will be resolved but not sure how long that will take (its been broken for 3-4 months)

Reply:
seems like a fix for this is imminent https://github.com/bazelbuild/rules_pkg/pull/484

Reply:
the hack for this has been removed, but currently it means using the above PR

i guess we will need to wait for it to land, but at least the hack is gone